File: LeetCode/Python/0268. Missing Number #2.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = max(nums)
        s = sum(nums)
        diff = (((n ** 2) + n) // 2) - s
        if not diff:
            if len(nums) == n:
                return 0
            else:
                return n + 1
        else:
            return diff

File: LeetCode/Java/0268. Missing Number #2.java
class Solution {
    public int missingNumber(int[] nums) {
        int max = 0;
        int sum = 0;
        for (int num: nums) {
            max = Math.max(max, num);
            sum += num;
        }
        
        int diff = (((max + 1) * max) / 2) - sum;
        if (diff != 0)
            return diff;
        if (max == nums.length)
            return 0;
        return max + 1;
    }
}
